src/App.tsx

```diff
@@ -1,28 +1,43 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { getFilteredTodos } from './utils/getFilterdTodos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { TypeError } from './types/TypeError';
+import { SelectedType } from './types/SelectedType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<TypeError>(
+    TypeError.DEFAULT,
+  );
+  const [selectedOption, setSelectedOption] = useState<SelectedType>(
+    SelectedType.ALL,
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(TypeError.LOADING));
+  }, []);
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.

Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.